### ISSUE TITLE
N816 doesn't flag lower-case variable names

### ISSUE THREAD

**Issue by isaacag** [astral-sh/ruff#14905]
Context managers are not taken into consideration for variable naming conventions, despite PEP8 not making exceptions for them. The only way to trigger variable rules is to wrap them in other structures like a `def`.

```python
def example_function():
    Variable = 10 # N806 triggered
    with open('file.txt') as file:
        Variable = file.read() # N806 triggered
```

```python
with open('file.txt') as file:
    Variable = file.read() # No rule triggered
```

**Comment by MichaReiser** [comment]
The current behavior makes sense to me, considering that the rule is about variable names in functions and doesn't enforce lower case variable names in general. 

> Checks for the use of non-lowercase variable names in functions.

And `Variable` in the second example is a variable in the module scope.

**Comment by isaacag** [comment]
If possible I think it would be nice to have an additional rule covering those cases, even if is another scope.

**Comment by MichaReiser** [comment]
> If possible I think it would be nice to have an additional rule covering those cases, even if is another scope.

There's https://docs.astral.sh/ruff/rules/mixed-case-variable-in-global-scope/ but it doesn't forbid using variable names that start with an uppercase character. I'm not sure what the motivation for it is but the rule matches the behavior of the upstream rule https://github.com/PyCQA/pep8-naming/pull/69

**Comment by isaacag** [comment]
That is odd, docs leave the impression that any non snake case variables will trigger the rule.

**Comment by MichaReiser** [comment]
I merge this with https://github.com/astral-sh/ruff/issues/15069 that lists a few reasons
